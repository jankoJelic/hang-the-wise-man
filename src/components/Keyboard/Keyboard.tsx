import CONSTANTS from "constants";
import React from "react";

const Keyboard = () => {
  const alphabet = "abcdefghijklmnopqrstuvwxyz";

  return (
    <div className="flex flex-wrap p-4 mx-4">
      {alphabet.split("").map((letter) => (
        <div className="flex w-8 aspect-square rounded-lg m-4 cursor-pointer bg-mainExtraLight items-center justify-center">
          {letter}
        </div>
      ))}
    </div>
  );
};

export default Keyboard;
