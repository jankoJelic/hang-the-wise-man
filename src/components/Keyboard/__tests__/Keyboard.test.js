import Keyboard from "../Keyboard";
import renderer from "react-test-renderer";
import { fireEvent, render, screen } from "@testing-library/react";

const mockClickLetter = jest.fn();
const mockUsedLetters = "ASDF";

const renderKeyboard = () =>
  render(
    <Keyboard onClickLetter={mockClickLetter} usedLetters={mockUsedLetters} />
  );

it("renders correctly", () => {
  const tree = renderer
    .create(
      <Keyboard onClickLetter={mockClickLetter} usedLetters={mockUsedLetters} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("contains 26 letters", () => {
  renderKeyboard();

  expect(screen.getAllByRole("letter")).toHaveLength(26);
});

it("fires onClickLetter when a letter is clicked", () => {
  renderKeyboard();

  const letter = screen.getByText("X");
  fireEvent.click(letter);

  expect(mockClickLetter).toHaveBeenCalledTimes(1);
});
