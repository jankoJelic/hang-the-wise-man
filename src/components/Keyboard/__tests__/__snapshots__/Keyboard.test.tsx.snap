// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="flex flex-wrap p-4 mx-4 mt-8 md:px-16"
>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-transparent
      "
    onClick={[Function]}
    role="letter"
  >
    A
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    B
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    C
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-transparent
      "
    onClick={[Function]}
    role="letter"
  >
    D
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    E
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-transparent
      "
    onClick={[Function]}
    role="letter"
  >
    F
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    G
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    H
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    I
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    J
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    K
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    L
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    M
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    N
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    O
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    P
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    Q
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    R
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-transparent
      "
    onClick={[Function]}
    role="letter"
  >
    S
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    T
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    U
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    V
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    W
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    X
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    Y
  </div>
  <div
    className="flex w-8 aspect-square rounded-lg m-2 
      cursor-pointer items-center justify-center
      bg-mainExtraLight
      shadow-sm"
    onClick={[Function]}
    role="letter"
  >
    Z
  </div>
</div>
`;
